draft-ietf-tls-esni.md

```diff
@@ -327,20 +374,23 @@ then it sends the following structure in EncryptedExtensions:
 ~~~
 
 retry_configs
-: An ECHConfigs structure containing one or more ECHConfig structures in
-decreasing order of preference that the client should use on subsequent
-connections to encrypt the ClientHelloInner structure.
+: An ECHConfigs structure containing one or more ECHConfig structures, in
+decreasing order of preference, to be used by the client in subsequent
+connection attempts.
 
-This protocol also defines the "ech_required" alert, which is sent by the client
-when it offered an "encrypted_client_hello" extension which was not accepted by
+This document also defines the "ech_required" alert, which is sent by the client
+when it offered an "encrypted_client_hello" extension that was not accepted by
 the server.
+[[TODO: Is this a "MAY", "SHOULD", or "MUST"?]]
+[[TODO: Add a reference to a point in the document that provides more context.]]
 
 # The "ech_nonce" extension {#ech-nonce}
 
-When using ECH, the client MUST also add an extension of type "ech_nonce" to the
-ClientHelloInner (but not to the outer ClientHello). This nonce ensures that the
-server's encrypted Certificate can only be read by the entity which sent this
-ClientHello. This extension is defined as follows:
+When using ECH, the client MUST include an extension of type "ech_nonce" to the
+ClientHelloInner (but not to the ClientHelloOuter). This nonce is used to ensure
+that the server's encrypted Certificate can only be read by the entity which
+sent this ClientHello. [[TODO: It's not clear from context what this means.]]
```

Comment:
Can we remove this TODO and make it an issue?

Reply:
This is addressed by https://github.com/tlswg/draft-ietf-tls-esni/issues/253. But I think it might be worth resolving this issue as part of this PR. Or should we wait?

Reply:
Let's wait.

Reply:
Ack, removing TODO.